src/Safebox.sol

```diff
@@ -0,0 +1,238 @@
+     * @param data The new value of the parameter.
+     */
+    event File(bytes32 indexed what, address data);
+    /**
+     * @notice The recipient has been sucessfully changed.
+     * @param recipient The new recipient address.
+     */
+    event RecipientChange(address indexed recipient);
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "Safebox/not-authorized");
+        _;
+    }
+
+    /**
+     * @param _vat The MCD vat module.
+     * @param _owner The safebox owner.
+     * @param _custodian The safebox custodian.
+     * @param _recipient The recipient for assets in the safebox.
+     */
+    constructor(address _vat, address _owner, address _custodian, address _recipient) {
+        require(_recipient != address(0), "Safebox/invalid-recipient");
+
+        wards[_owner] = 1;
+        emit Rely(_owner);
+
+        can[_custodian] = 1;
+        emit Hope(_custodian);
+
+        vat = VatLike(_vat);
+        recipient = _recipient;
+    }
+
+    /*//////////////////////////////////
+            MakerDAO Interfaces
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` owner access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` owner access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"recipient"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "recipient") {
+            require(data != address(0), "Safebox/invalid-recipient");
+            pendingRecipient = data;
+            emit File(what, data);
+        } else {
+            revert("Safebox/file-unrecognized-param");
+        }
+    }
+
+    /**
+     * @notice Withdraws ERC20-like tokens from this contract.
+     * @dev Anyone can call this function after MakerDAO governance executes an Emergency Shutdown.
+     * @param token The token to be withdrawn.
+     * @param amount The amount of tokens.
+     */
+    function withdraw(address token, uint256 amount) external {
+        require(wards[msg.sender] == 1 || vat.live() == 0, "Safebox/not-owner");
+
+        _safeTransfer(token, recipient, amount);
+        emit Withdraw(token, amount);
+    }
+
+    /**
+     * @dev Handles `transfer` for tokens with non-standard ERC-20 implementations.
+     * See https://github.com/d-xo/weird-erc20
+     * @param to The destination address.
+     * @param amount The amount to be transfered.
+     */
+    function _safeTransfer(address token, address to, uint256 amount) internal {
```

Comment:
I'd reduce the verbosity of the code just using a simple `transfer`. I don't see Maker interacting with a token that doesn't `revert` and this contract is behind governance setting it up for a specific purpose. This is only useful for contracts that are open to any token.
At least I'd remove the internal function and just put the code in `withdraw` (same for `deposit`) as there is not reuse of it. But honestly I'd just keep it simple with the regular `transfer`.

Reply:
Makes sense, removed.

Reply:
Another thing I considered was to make `token` a constructor parameter instead of a parameter for `withdraw`.

As it is, the Safebox can be seen as multi-token, but there will probably never be a case for it to hold more than 1 asset.